diff --git a/sdk/android/BUILD.gn b/sdk/android/BUILD.gn
--- a/sdk/android/BUILD.gn
+++ b/sdk/android/BUILD.gn
@@ -341,6 +341,7 @@
       "api/org/webrtc/SessionDescription.java",
       "api/org/webrtc/StatsObserver.java",
       "api/org/webrtc/StatsReport.java",
+      "api/org/webrtc/TalkKeyRing.java",
       "api/org/webrtc/TurnCustomizer.java",
       "api/org/webrtc/VideoProcessor.java",
       "api/org/webrtc/VideoSource.java",
@@ -838,6 +839,7 @@
       "src/jni/pc/ssl_certificate_verifier_wrapper.h",
       "src/jni/pc/stats_observer.cc",
       "src/jni/pc/stats_observer.h",
+      "src/jni/pc/talk_key_ring.cc",
       "src/jni/pc/turn_customizer.cc",
       "src/jni/pc/turn_customizer.h",
     ]
@@ -902,6 +904,7 @@
       "../../api/video_codecs:video_codecs_api",
       "../../call:call_interfaces",
       "../../modules/audio_device",
+      "../../modules/talk_frame_crypto",
       "../../modules/utility",
       "../../p2p:port",
       "../../pc:media_stream_observer",
@@ -1592,6 +1595,7 @@
       "api/org/webrtc/SessionDescription.java",
       "api/org/webrtc/StatsObserver.java",
       "api/org/webrtc/StatsReport.java",
+      "api/org/webrtc/TalkKeyRing.java",
       "api/org/webrtc/TurnCustomizer.java",
     ]
     namespace = "webrtc::jni"
diff --git a/sdk/android/src/jni/DEPS b/sdk/android/src/jni/DEPS
--- a/sdk/android/src/jni/DEPS
+++ b/sdk/android/src/jni/DEPS
@@ -9,6 +9,7 @@
   "+modules/audio_device/include/audio_device.h",
   "+modules/audio_processing/include/audio_processing.h",
   "+modules/include",
+  "+modules/talk_frame_crypto",
   "+modules/utility/include/jvm_android.h",
   "+modules/video_coding",
   "+pc",
