diff --git a/p2p/client/basic_port_allocator.cc b/p2p/client/basic_port_allocator.cc
index ed9db392c4..f9fc769d9b 100644
--- a/p2p/client/basic_port_allocator.cc
+++ b/p2p/client/basic_port_allocator.cc
@@ -1361,6 +1361,13 @@ void AllocationSequence::DisableEquivalentPhases(const Network* network,
     *flags |= webrtc::PORTALLOCATOR_DISABLE_TCP;
   }
 
+  // Patched from https://bugs.chromium.org/p/webrtc/issues/detail?id=11328
+  // If the allocation is not finished for TCP and still active for that protocol
+  // we must disable the TCP port allocator to prevent a new allocation for this network.
+  if (state_ == kRunning && phase_ <= PHASE_TCP && !IsFlagSet(webrtc::PORTALLOCATOR_DISABLE_TCP)) {
+    *flags |= webrtc::PORTALLOCATOR_DISABLE_TCP;
+  }
+
   if (config_ && config) {
     // We need to regather srflx candidates if either of the following
     // conditions occurs:
