diff --git a/modules/BUILD.gn b/modules/BUILD.gn
index f9bb6b6d36..d616a46837 100644
--- a/modules/BUILD.gn
+++ b/modules/BUILD.gn
@@ -20,6 +20,7 @@ group("modules") {
     "remote_bitrate_estimator",
     "rtp_rtcp",
     "sframe",
+    "talk_frame_crypto",
     "utility",
     "video_coding",
   ]
diff --git a/sdk/BUILD.gn b/sdk/BUILD.gn
index 8b08459819..0e66b90bd5 100644
--- a/sdk/BUILD.gn
+++ b/sdk/BUILD.gn
@@ -1086,6 +1086,8 @@ if (is_ios || is_mac) {
         "objc/api/peerconnection/RTCStatisticsReport+Private.h",
         "objc/api/peerconnection/RTCStatisticsReport.h",
         "objc/api/peerconnection/RTCStatisticsReport.mm",
+        "objc/api/peerconnection/RTCTalkKeyRing.h",
+        "objc/api/peerconnection/RTCTalkKeyRing.mm",
         "objc/api/peerconnection/RTCTracing.h",
         "objc/api/peerconnection/RTCTracing.mm",
         "objc/api/peerconnection/RTCVideoTrack+Private.h",
@@ -1152,6 +1154,7 @@ if (is_ios || is_mac) {
         "../api/video_codecs:video_codecs_api",
         "../common_video",
         "../media:media_constants",
+        "../modules/talk_frame_crypto",
         "../modules/video_coding:video_codec_interface",
         "../pc:peer_connection_factory",
         "../pc:webrtc_sdp",
@@ -1208,6 +1211,7 @@ if (is_ios || is_mac) {
             "objc/unittests/RTCPeerConnectionFactory_xctest.m",
             "objc/unittests/RTCPeerConnectionTest.mm",
             "objc/unittests/RTCSessionDescriptionTest.mm",
+            "objc/unittests/RTCTalkKeyRingTest.mm",
             "objc/unittests/RTCTracingTest.mm",
             "objc/unittests/frame_buffer_helpers.h",
             "objc/unittests/frame_buffer_helpers.mm",
@@ -1257,6 +1261,7 @@ if (is_ios || is_mac) {
             "../common_video",
             "../media:codec",
             "../media:rtc_media_tests_utils",
+            "../modules/talk_frame_crypto:web_test_vectors",
             "../modules/video_coding:video_codec_interface",
             "../rtc_base:refcount",
             "../rtc_base:rtc_event",
@@ -1404,6 +1409,7 @@ if (is_ios || is_mac) {
           "objc/api/peerconnection/RTCSSLAdapter.h",
           "objc/api/peerconnection/RTCSessionDescription.h",
           "objc/api/peerconnection/RTCStatisticsReport.h",
+          "objc/api/peerconnection/RTCTalkKeyRing.h",
           "objc/api/peerconnection/RTCTracing.h",
           "objc/api/peerconnection/RTCCertificate.h",
           "objc/api/peerconnection/RTCCryptoOptions.h",
diff --git a/sdk/objc/DEPS b/sdk/objc/DEPS
index 4cff92caf1..b548592476 100644
--- a/sdk/objc/DEPS
+++ b/sdk/objc/DEPS
@@ -8,6 +8,7 @@ include_rules = [
   "+common_video/libyuv/include",
   "+logging/rtc_event_log/rtc_event_log_factory.h",
   "+media",
+  "+modules/talk_frame_crypto",
   "+modules/video_coding",
   "+pc",
   "+system_wrappers",
